### PR TITLE
passkeys: documentation

### LINKED ISSUE #23660: passkeys: documentation
### Description

Documentation for passkeys. Maybe also blog post can be good (not sure if after Keycloak 23 release or before).

### PR DESCRIPTION
**Pull request by tnorimat**
closes #23660

### CODE REVIEW

**Comment by tnorimat** [comment]
@mposolda Could you review the PR?

File: docs/documentation/server_admin/topics/authentication/passkeys.adoc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[id="passkeys_{context}"]
+=== Passkeys
+
+{project_name} provides support for https://fidoalliance.org/passkeys/[Passkeys]. {project_name} works as a Passkeys' https://passkeys.dev/docs/reference/terms/#relying-party-rp[Relying Party (RP)].
```
Comment (mposolda) [inline docs/documentation/server_admin/topics/authentication/passkeys.adoc:4]:
```suggestion
{project_name} provides preview support for https://fidoalliance.org/passkeys/[Passkeys]. {project_name} works as a Passkeys' https://passkeys.dev/docs/reference/terms/#relying-party-rp[Relying Party (RP)].
```

**Review by mposolda — COMMENTED** [review]
@tnorimat Thanks a lot!

I've added just 2 notes inline. I would be a bit more defensive and would like to rather add something about "preview support" for now (even if we don't have preview features flag), as we don't yet have automated tests for passkeys, and overally with preview, we have more freedom to possibly update things later if needed etc. Does it work for you?

**Comment by mposolda** [comment]
@tnorimat There are also some issues with the external links. Can you please check it? If links are problematic, I think it is possible to somehow disable checking the link for them in the "external links" test.

File: docs/documentation/release_notes/topics/23_0_0.adoc [hunk 2]
```diff
@@ -13,3 +13,13 @@ the latest FAPI 2 draft specifications when communicating with your clients. Tha
 
 Keycloak has preview for support for OAuth 2.0 Demonstrating Proof-of-Possession at the Application Layer (DPoP). Thanks to
 https://github.com/tnorimat[Takashi Norimatsu] and https://github.com/dteleguin[Dmitry Telegin] for their contributions.
+
+= Passkeys support
+
+Keycloak has supported https://fidoalliance.org/passkeys/[Passkeys].
```
[the remark this answers is not in the document]
Comment (tnorimat) [reply docs/documentation/release_notes/topics/23_0_0.adoc]: I fixed the point.

File: docs/documentation/server_admin/topics/authentication/passkeys.adoc (hunk 1 above)
Comment (tnorimat) [reply docs/documentation/server_admin/topics/authentication/passkeys.adoc]: I fixed the point.

**Comment by tnorimat** [comment]
@mposolda I fixed the points you commented.

File: docs/documentation/release_notes/topics/23_0_0.adoc [hunk 3]
```diff
@@ -13,3 +13,13 @@ the latest FAPI 2 draft specifications when communicating with your clients. Tha
...
+
+= Passkeys support
+
+Keycloak has preview support for https://fidoalliance.org/passkeys/[Passkeys].
+
+Passkey registration and authentication are realized by the features of WebAuthn.
+Therefore, users of Keycloak can do passkey registration and authentication by existing WebAuthn registraton and authentication.
+
+Both synced passkeys and device-bound passkeys can be used for both Same-Device and Cross-Device Authentication.
+However, passkeys' operations success depends on the user's environment. Make sure which operations can success in https://passkeys.dev/device-support/[the environment].
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/23_0_0.adoc:25]:
```suggestion
However, passkeys operations success depends on the user's environment. Make sure which operations can succeed in https://passkeys.dev/device-support/[the environment].
```

File: docs/documentation/server_admin/topics/authentication/passkeys.adoc [hunk 4]
```diff
@@ -0,0 +1,13 @@
+[id="passkeys_{context}"]
+=== Passkeys
+
+{project_name} provides preview support for https://fidoalliance.org/passkeys/[Passkeys]. {project_name} works as a Passkeys' Relying Party (RP).
```
Comment (andymunro) [inline docs/documentation/server_admin/topics/authentication/passkeys.adoc:4]:
```suggestion
{project_name} provides preview support for https://fidoalliance.org/passkeys/[Passkeys]. {project_name} works as a Passkeys Relying Party (RP).
```

**Review by mposolda — DISMISSED** [review]
@tnorimat Thanks!

File: docs/documentation/server_admin/topics/authentication/passkeys.adoc [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+Passkey registration and authentication are realized by the features of xref:webauthn_{context}[WebAuthn].
+Therefore, users of {project_name} can do passkey registration and authentication by existing xref:webauthn_{context}[WebAuthn registraton and authentication].
+
+[NOTE]
+====
+Both synced passkeys and device-bound passkeys can be used for both Same-Device and Cross-Device Authentication (CDA).
+However, passkeys' operations success depends on the user's environment. Make sure which operations can success in https://passkeys.dev/device-support/[the environment].
```
Comment (andymunro) [inline docs/documentation/server_admin/topics/authentication/passkeys.adoc:12]:
```suggestion
However, passkeys operations success depends on the user's environment. Make sure which operations can success in https://passkeys.dev/device-support/[the environment].
```

**Review by andymunro — CHANGES_REQUESTED** [review]
Looks good. Just a few changes to suggest.

File: docs/documentation/release_notes/topics/23_0_0.adoc (hunk 3 above)
Comment (tnorimat) [reply docs/documentation/release_notes/topics/23_0_0.adoc]: I fixed the point.

File: docs/documentation/server_admin/topics/authentication/passkeys.adoc (hunk 4 above)
Comment (tnorimat) [reply docs/documentation/server_admin/topics/authentication/passkeys.adoc]: I fixed the point.

**Comment by tnorimat** [comment]
@andymunro I fixed the points you commented. Thank you.

**Review by mposolda — APPROVED** [review]
@tnorimat Thanks!

**Review by mposolda — COMMENTED** [review]
@andymunro Thank you for the review!

**Pull request merged**